File: tests/fixtures/code_samples/sample.cpp
/**
 * Sample C++ code for testing code parsing.
 */

#include <string>

namespace math {

/**
 * Calculate power
 */
int power(int base, int exp) {
    int result = 1;
    for (int i = 0; i < exp; i++) {
        result *= base;
    }
    return result;
}

}

/**
 * Vector class
 */
class Vector {
private:
    double x, y;

public:
    /**
     * Constructor
     */
    Vector(double x, double y) : x(x), y(y) {}

    /**
     * Get magnitude
     */
    double magnitude() {
        return sqrt(x*x + y*y);
    }
};

File: tests/fixtures/code_samples/sample.rs
/// Sample Rust module for testing code parsing

pub struct Point {
    x: i32,
    y: i32,
}

pub enum Color {
    Red,
    Green,
    Blue,
}

/// Calculate distance between two points
pub fn distance(p1: &Point, p2: &Point) -> f64 {
    let dx = (p2.x - p1.x) as f64;
    let dy = (p2.y - p1.y) as f64;
    (dx * dx + dy * dy).sqrt()
}

impl Point {
    /// Create a new point
    pub fn new(x: i32, y: i32) -> Self {
        Point { x, y }
    }

    /// Move point by offset
    pub fn translate(&mut self, dx: i32, dy: i32) {
        self.x += dx;
        self.y += dy;
    }
}
